src/runtimes/docker/utils/getNativeArchitecture.ts

```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+
+export type CpuArchitecture =
+    | 'amd64'
+    | '386'
+    | 'arm64'
+    | 'arm'
+    | 'mips'
+    | 'mipsle'
+    | 'ppc64'
+    | 's390x';
+
+/**
+ * Returns native architecture of the current machine
+ * @returns native architecture of the current machine
+ */
+export function getNativeArchitecture(): CpuArchitecture {
+    switch (os.arch()) {
+        case 'arm':
+            return 'arm';
+        case 'arm64':
+            return 'arm64';
+        case 'ia32':
+            return '386';
+        case 'mips':
+            return 'mips';
+        case 'mipsel':
+            return 'mipsle';
+        case 'ppc':
+        case 'ppc64':
+            return 'ppc64';
+        case 's390x':
+            return 's390x';
+        case 'x64':
+        default:
+            return 'amd64';
+    }
```

Comment:
A shorter way to write this that's also potentially more robust:

```suggestion
    const arch = os.arch();
    switch (arch) {
        case 'ia32':
            return '386';
        case 'x64':
            return 'amd64';
        default:
            return arch;
    }
```

Reply:
Don't know if i agree with this, the switch statement was based on mapping https://nodejs.org/api/os.html#osarch to https://go.dev/doc/install/source#environment. However, I do think it could be shorter and more robust, so something like this?

```typescript
   const arch = os.arch() || 'amd64';
   switch (arch) {
        case 'ia32':
            return '386';
        case 'mipsel':
            return 'mipsle';
        case 'ppc':
            return 'ppc64';
        case 'x64':
            return 'amd64'
        default:
            return os;
    }
```

Reply:
Yes, I missed that the `mipsel` -> `mipsle` is intentional (though we should validate that in the Go code and make sure it's not just a documentation typo, and if it is indeed not a typo, leave an explanatory comment), and `ppc` -> `ppc64`.